### PR TITLE
Allow more action definition types

### PR DESCRIPTION
**Pull request by dpikt**

### CODE REVIEW

**Comment by dpikt** [comment]
@davemcorwin ready for review!

File: src/middleware.js [hunk 1]
```diff
@@ -65,23 +70,47 @@ export default function ({ onUnauthorized, ...options }) {
...
+            payload: { ...rest, response },
+          }))
+        }
+        
+      })
   }
 }
+
+// Create an action from an action "definition."
+const parseAction = ({ action, payload={}, error=false }) => {
```
Comment (davemcorwin) [inline src/middleware.js]: I'm not sure that setting the defaults for destructured values like this works. I had some issues with this which I have the more explicit merging of options up above.

File: src/middleware.js [hunk 2]
```diff
@@ -65,23 +70,47 @@ export default function ({ onUnauthorized, ...options }) {
...
+  // If it's an action creator, create the action
+  if (typeof action === 'function') return action(payload.response)
```
Comment (davemcorwin) [inline src/middleware.js]: Thoughts on using `switch` here? I'm good either way.

File: src/middleware.js [hunk 3]
```diff
@@ -43,17 +43,22 @@ export default function ({ onUnauthorized, ...options }) {
...
       mode,
+      actions,
       types,
       url,
       ...rest,
     } = lpApi
 
-    const [ requestType, successType, errorType ] = types
+    // Alias 'actions' with 'types' for backwards compatibility
+    const [ requestAction, successAction, errorAction ] = actions || types
```
Comment (davemcorwin) [inline src/middleware.js]: yay! Why not parse them all upfront?

Comment (dpikt) [inline src/middleware.js]: They all have different payloads, some of which depend on server responses.

File: src/middleware.js (hunk 1 above)
Comment (dpikt) [inline src/middleware.js]: It seemed to be working for me. What issues were you running into?

File: src/middleware.js (hunk 2 above)
Comment (dpikt) [inline src/middleware.js]: I was on the fence about it. It might be cleaner though.

**Comment by dpikt** [comment]
@davemcorwin responded!

File: src/middleware.js (hunk 3 above)
Comment (davemcorwin) [inline src/middleware.js]: good point. i suppose you could 'parse' them into a generic action creator that takes the payload and error status, then use that going forward. Not sure if that makes things simpler or more complicated. Good w me either way.

File: src/middleware.js (hunk 1 above)
Comment (davemcorwin) [inline src/middleware.js]: hmm, just tested in node repl and seems fine (so much nicer now that these features are actually in node...) I think my issue was due to either removing keys for undefined values (that I am now handling with `omitUndefined`) or needing an explicit `false` value or something. Whatevs, looks good, my bad.

**Pull request merged**